### PR TITLE
#19: Added a configurable list for excluding files from validation

### LINKED ISSUE #19: Add exclude files
add a configurable list of excluded files or modules.

**Use Case:**
Some specific repositories do not need a dependency check.

### PR DESCRIPTION
**Pull request by jakobbraun**
Closes #19

### CODE REVIEW

File: src/main/java/com/exasol/projectkeeper/validators/DeletedFilesValidator.java [hunk 1]
```diff
@@ -19,20 +20,27 @@
...
+        this.excludedFiles = excludedFiles;
     }
 
     @Override
     public Validator validate(final Consumer<ValidationFinding> findingConsumer) {
         FILES_THAT_MUST_NOT_EXIST.forEach((fileThatMustNotExist, reason) -> {
-            final File file = this.projectDirectory.resolve(fileThatMustNotExist).toFile();
+            final Path pathThatMustExist = this.projectDirectory.resolve(fileThatMustNotExist);
+            if (this.excludedFiles.isFileExcluded(this.projectDirectory.relativize(pathThatMustExist))) {
+                return;
```
Comment (AnastasiiaSergienko) [inline src/main/java/com/exasol/projectkeeper/validators/DeletedFilesValidator.java:41]: Does it return null here?

**Review by AnastasiiaSergienko — APPROVED** [review]
Approved with a couple of questions

File: pom.xml [hunk 2]
```diff
@@ -103,16 +103,23 @@
...
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <version>4.13.1</version>
+            <scope>test</scope>
+        </dependency>
         <dependency>
             <groupId>org.testcontainers</groupId>
             <artifactId>testcontainers</artifactId>
-            <version>1.14.3</version>
+            <version>1.15.0</version>
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply pom.xml:116]: I use them for testing the plugin. I run a plain container with maven, create a project and run in there the fixes.
I could not find an other solution to test the plugin yet.

File: pom.xml [hunk 3]
```diff
@@ -404,7 +411,7 @@
             <plugin>
                 <groupId>com.exasol</groupId>
                 <artifactId>project-keeper-maven-plugin</artifactId>
-                <version>0.3.0</version>
+                <version>${version}</version>
```
[the remark this answers is not in the document]
Comment (jakobbraun) [reply pom.xml:414]: It has no execution. Maven does not check the plugins at compile-time, but when they are executed.
In the travis.yml this is first installed and then called manually.

File: src/main/java/com/exasol/projectkeeper/validators/DeletedFilesValidator.java (hunk 1 above)
Comment (jakobbraun) [reply src/main/java/com/exasol/projectkeeper/validators/DeletedFilesValidator.java:41]: The return returns from the lambda in line 38, which is void

**Pull request merged**